snuba/web/delete_query.py

```diff
@@ -73,6 +84,33 @@ def delete_from_storage(
 
     results: dict[str, Result] = {}
     attr_info = _get_attribution_info(attribution_info)
+
+    # fail if too many mutations ongoing
+    cluster = storage.get_cluster()
+    if cluster.is_single_node():
+        query = f"""
+    SELECT count() as cnt
+    FROM system.mutations
+    WHERE table IN ({", ".join(map(repr, delete_settings.tables))}) AND is_done=0
+"""
+    else:
+        query = f"""
+SELECT max(cnt)
+FROM (
+    SELECT hostname() as host, count() as cnt
+    FROM clusterAllReplicas('{cluster.get_clickhouse_cluster_name()}', 'system', mutations)
+    WHERE table IN ({", ".join(map(repr, delete_settings.tables))}) AND is_done=0
```

Comment:
This will combine the counts for the tables right? Can we group by table and check the max of the results? So the max of the max for the cluster query, and it would be the max of the single node

Reply:
but if there are two tables for a node it will combine them

```
SELECT count(), table
FROM system.mutations
WHERE table in ('outcomes_raw_local', 'outcomes_hourly_local')
GROUP BY table
```
<img width="1171" alt="Screenshot 2024-09-27 at 2 29 02 PM" src="https://github.com/user-attachments/assets/a764f2d0-b9cb-4b01-97ff-08bf507a2188">

If you don't group by table, the total count is 19

This is for one node. This won't be a problem for the search issues table, but I'm just saying if we ever need to delete from multiple tables it will combine the counts

Reply:
Oh I see what you mean now, fixed.